ts/features/messages/components/Home/PagerViewContainer.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useCallback } from "react";
+import { NativeSyntheticEvent } from "react-native";
+import PagerView from "react-native-pager-view";
+import { OnPageSelectedEventData } from "react-native-pager-view/lib/typescript/PagerViewNativeComponent";
+import { IOStyles } from "@pagopa/io-app-design-system";
+import { useIODispatch, useIOStore } from "../../../../store/hooks";
+import { setShownMessageCategoryAction } from "../../store/actions";
+import { MessageList } from "./MessageList";
+import {
+  getMessagesViewPagerInitialPageIndex,
+  messageViewPageIndexToListCategory
+} from "./homeUtils";
+
+export const PagerViewContainer = React.forwardRef<PagerView>((_, ref) => {
+  const dispatch = useIODispatch();
+  const store = useIOStore();
+  const state = store.getState();
+  const initialPageIndex = getMessagesViewPagerInitialPageIndex(state);
```

Comment:
After logging, I noticed that the `useSelector` hook caused some rerenders when switching tabs, while your implementation didn't -- good catch!